lib/app.js

```diff
@@ -100,100 +100,193 @@ async.series([
-
-    program
-      .command('start [app]')
-      .description(t('matrix.help_start'))
-      .option('-a, --all')
-      .action(function (app, options) {
-        if (_.isUndefined(app)) {
-          console.log('\n> matrix start ¬\n');
-          console.log('\t    matrix start <app> -', t('matrix.start.help', {app: '<app>'}).grey)
-          endIt();
-        } else {
-          Matrix.validate.user(); //Make sure the user has logged in
-          Matrix.validate.device(); //Make sure the user has logged in
-          if (options.all) {
-            //FIXME: hacky
-            app = 'all-applications'
-          }
-          Matrix.api.device.setId(Matrix.config.device.identifier);
-          console.log(t('matrix.start.starting_app') + ': ', app, Matrix.config.device.identifier);
-          Matrix.loader.start();
-          Matrix.api.app.start(app, Matrix.config.device.identifier, function (err, res) {
-            Matrix.loader.stop();
-            if (!err) {
-              console.log(t('matrix.start.start_app_successfully') + ': ', app);
-            } else {  
-              console.log(t('matrix.start.start_app_error') + ':', app , ' (' + err.message.red + ')');
-            } 
-            endIt();
+  program.command('Apps');
+  program.command('search', t('matrix.help_search').grey);
+  program.command('install', t('matrix.help_install') + '\n[ app / sensor ] <name>. '.grey + t('matrix.help_install_default').grey)
+  program.command('uninstall', t('matrix.help_uninstall'), ' <app>')
+  program.command('config', t('matrix.help_config'))
+  program.command('update', t('matrix.help_update') + '\n<appName> <version>'.grey)
+
+  //Start command
+  program
+    .command('start [app]')
+    .description(t('matrix.help_start'))
+    .option('-a, --all')
+    .action(function (app, options) {
+      //Make sure the user has logged in
+      Matrix.validate.user();
+      Matrix.validate.device();
+
+      Matrix.api.device.setId(Matrix.config.device.identifier);
+      console.log(t('matrix.start.starting_app') + ': ', app, Matrix.config.device.identifier);
+
+      Matrix.firebaseInit(function () {
+        //Get the app id for name
+        Matrix.firebase.app.getIDForName( app, function(err, appId){
+          if (err) return console.error(err);
+          debug('appId>', appId);
+          //Get the current status of app
+          Matrix.firebase.app.getStatus(appId, function (status) {
+            debug("Get current status: " + Matrix.config.user.id + '>' + Matrix.config.device.identifier + '>' + appId + '>' + status);
+
+            if (_.isUndefined(app)) {
+              console.log('\n> matrix start ¬\n');
+              console.log('\t    matrix start <app> -', t('matrix.start.help', {app: '<app>'}).grey)
+              endIt();
+            //If the status of the app is different of active or error doesn't execute de start command
+            } else if(status !== 'inactive' && status !== 'error' ){
```

Comment:
This is bad logic. If an app is pending, it will throw this error. If `runtime` doesn't exist it will throw this error.  This should be `status === 'active'`